Sources/Datadog/Tracing/Integrations/TracingWithLoggingIntegration.swift

```diff
@@ -31,13 +31,13 @@ internal struct TracingWithLoggingIntegration {
     }
 
     /// `DatadogCore` instance managing this integration.
-    let core: DatadogCoreProtocol
+    weak var core: DatadogCoreProtocol?
```

Comment:
Is it unnecessarily strong in the current implementation architecture or the new architecture changed how it's need to be handled here?